base/inference.jl

```diff
@@ -1269,7 +1269,6 @@ function tmerge(typea::ANY, typeb::ANY)
         if isa(typea, DataType) && isa(typeb, DataType) && length(typea.parameters) == length(typeb.parameters) && !isvatuple(typea) && !isvatuple(typeb)
             return typejoin(typea, typeb)
         end
-        return Tuple
```

Comment:
did something change such that the fix for https://github.com/JuliaLang/julia/issues/6704 isn't useful anymore? Tuples must widen really fast or inference can get stuck trying to infer some really bad cases. (IIUC, you are finding that the lack of a recursive version of this test causes https://github.com/JuliaLang/julia/issues/14743#issuecomment-229737133)